### ISSUE TITLE
DB error: failed to download vulnerability DB: OCI artifact error: OCI artifact error: OCI repository error: GET 

### ISSUE THREAD

**Issue by maximiliarthur** [aquasecurity/trivy-action#107]
## Description
I was running the trivy on the github action workflow and give an error message that the vulnerability database can't be downloaded.

## What did you expect to happen?

The trivy can download the vulnerability database

## What happened instead?

The trivy gives an unknown error when failed the download

## Output of run with `-debug`:

```
Global options:  
20[22](https://github.com/payfazz/post-push-notif/runs/6155553216?check_suite_focus=true#step:9:22)-04-25T10:16:03.494Z	INFO	Need to update DB
2022-04-25T10:16:03.494Z	INFO	DB Repository: ghcr.io/aquasecurity/trivy-db
2022-04-25T10:16:03.494Z	INFO	Downloading DB...
2022-04-[25](https://github.com/payfazz/post-push-notif/runs/6155553216?check_suite_focus=true#step:9:25)T10:27:57.307Z	FATAL	DB error: failed to download vulnerability DB: OCI artifact error: OCI artifact error: OCI repository error: GET https://ghcr.io/v2/aquasecurity/trivy-db/manifests/2: UNKNOWN: unknown error
```

**Comment by simar7** [comment]
hi @maximiliarthur - that's strange indeed. Was this a one time thing (networking issues, etc.) or you are able to constantly reproduce this?

**Comment by maximiliarthur** [comment]
yeah actually I have tried to re-run the jobs but still reproduce this error

**Comment by maximiliarthur** [comment]
does the trivy official action support scan the local images? like this one
```
docker run --rm -v /var/run/docker.sock:/var/run/docker.sock \
    -v $HOME/Library/Caches:/root/.cache/ aquasec/trivy:0.18.3 python:3.4-alpine

```

**Comment by simar7** [comment]
I see. I was digging around if this issue was already reported and I think it might be related to this: https://github.com/aquasecurity/trivy/issues/1877

**Comment by maximiliarthur** [comment]
Yeah, I've read the issue first before but this is a different environment I think. Because I ran the trivy as a workflow using the trivy actions

**Comment by maximiliarthur** [comment]
Because I've implemented to other repository and no issue occurs. But, only the certain repository that produces this issue

**Comment by arareko** [comment]
I've constantly been experiencing this since the end of last week. I updated my GH worklow to skip trivy setup on subsequent executions of Trivy during the same job but this also happens on the first time execution.

**Comment by anbha15** [comment]
Got below issue. looks like rate limit issue. recently started for last few days intermittently getting this error.
2024-10-31T10:04:18Z	INFO	[vulndb] Need to update DB
2024-10-31T10:04:18Z	INFO	[vulndb] Downloading vulnerability DB...
2024-10-31T10:04:18Z	INFO	[vulndb] Downloading artifact...	repo="ghcr.io/aquasecurity/trivy-db:2"
0 B / 54.90 MiB [____________________________________________________________________] 0.00% ? p/s ?320.00 KiB / 54.90 MiB [>____________________________________________________________] 0.57% ? p/s ?54.90 MiB / 54.90 MiB [----------------------------------------------------------->] 100.00% ? p/s ?54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 90.41 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 90.41 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 90.41 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 84.58 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 84.58 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 84.58 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 79.12 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 79.12 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [---------------------------------------------->] 100.00% 79.12 MiB p/s ETA 0s54.90 MiB / 54.90 MiB [-------------------------------------------------] 100.00% 23.28 MiB p/s 2.6s2024-10-31T10:04:21Z	INFO	[vulndb] Artifact successfully downloaded	repo="ghcr.io/aquasecurity/trivy-db:2"
2024-10-31T10:04:21Z	INFO	[vuln] Vulnerability scanning is enabled
2024-10-31T10:04:21Z	INFO	[secret] Secret scanning is enabled
2024-10-31T10:04:21Z	INFO	[secret] If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-10-31T10:04:21Z	INFO	[secret] Please see also https://aquasecurity.github.io/trivy/v0.56/docs/scanner/secret#recommendation for faster secret detection
2024-10-31T10:04:30Z	INFO	[javadb] Downloading Java DB...
2024-10-31T10:04:30Z	INFO	[javadb] Downloading artifact...	repo="ghcr.io/aquasecurity/trivy-java-db:1"
2024-10-31T10:04:30Z	ERROR	[javadb] Failed to download artifact	repo="ghcr.io/aquasecurity/trivy-java-db:1" err="oci download error: failed to fetch the layer: GET https://ghcr.io/v2/aquasecurity/trivy-java-db/blobs/sha256:a99d0591d8eccbde4d57a6307ad3ba5e7234cfd96f5b8f0622c516dd2a00364d: TOOMANYREQUESTS: retry-after: 779.569µs, allowed: 44000/minute"
2024-10-31T10:04:31Z	FATAL	Fatal error	image scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:a445662b25b439747c7b9c01a09f6881addf77a7b0f81e2f3d6bef59412f03f9): post analysis error: post analysis error: Unable to initialize the Java DB: Java DB update failed: OCI artifact error: failed to download Java DB: failed to download artifact from any source
Error: Process completed with exit code 1.

**Comment by Agnes4Him** [comment]
I think these are all rate-limit related. I'm experiencing the same thing.

**Comment by ed-boykin** [comment]
Yup, same here. I was getting the error, then it went away for an hour or so and now its back.
